include/mbedtls/pk.h

```diff
@@ -248,13 +350,24 @@ static inline size_t mbedtls_pk_get_len( const mbedtls_pk_context *ctx )
 }
 
 /**
- * \brief           Tell if a context can do the operation given by type
+ * \brief           Tell if a context can do the operations given by type
+ *
+ * \note            This function can be used to identify the type of key
+ *                  (e.g. RSA vs ECC), and a superset of permitted
+ *                  operations. It is possible that this function returns
+ *                  true but some operations are not allowed. For example
+ *                  this function always returns true if ctx is an RSA
+ *                  context and type is MBEDTLS_PK_RSA, but the key may
+ *                  be restricted to any subset of operations among signature,
+ *                  verification, encryption and decryption. To determine
+ *                  which operations a key allow, attempt the operation and
+ *                  check the return status.
  *
  * \param ctx       Context to test
  * \param type      Target type
  *
- * \return          0 if context can't do the operations,
- *                  1 otherwise.
+ * \return          1 if context can do the operations,
+ *                  0 otherwise.
  */
 int mbedtls_pk_can_do( const mbedtls_pk_context *ctx, mbedtls_pk_type_t type );
```

Comment:
It's “can do”, not “may do”. This function can't tell “may do” because it doesn't have enough information: it doesn't know what the user wants to do. For example, an RSA key whose policy is “only for PSS signature” can't do encryption or PKCS#1v1.5 signature, but there's no way to ask “May I do PSS signature?” in the current API.

A future version will have some kind of “may do” function. I don't know yet what shape it'll take.